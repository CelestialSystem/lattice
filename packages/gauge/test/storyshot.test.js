import { mount } from 'enzyme';
import RadialGaugeComponent from '../src/index.js';
import React from 'react';

describe('To test the Radial Gauge Component.', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<RadialGaugeComponent value={50} />);
  });

  it('test componentDidMount', () => {
    wrapper.instance().componentDidMount();
    expect(wrapper.instance().gauge.value).toEqual(50);
  });

  it('test componentWillReceiveProps', () => {
    expect(wrapper.instance().gauge.value).toEqual(50);
    wrapper.instance().componentWillReceiveProps({ value: 50 });
    expect(wrapper.instance().gauge.value).toEqual(50);

    wrapper.instance().componentWillReceiveProps({ value: 10 });
    expect(wrapper.instance().gauge.value).toEqual(10);
  });
});
