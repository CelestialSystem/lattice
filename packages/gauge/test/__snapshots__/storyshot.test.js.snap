// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots gauge basic 1`] = `
<div
  style={
    Object {
      "height": "100vh",
      "padding": 12,
      "width": "100vw",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexWrap": "wrap",
        "justifyContent": "space-between",
      }
    }
  >
    <div
      className=""
    >
      <svg
        height={200}
        width={200}
      >
        <g
          className="lattice__gauge"
          style={
            Object {
              "transform": "translate(100px, 100px)",
            }
          }
        >
          <path
            className="Gauge-gaugeColor1-2"
            d="M-89.97187646481323,-2.249765632324131A90,90,0,0,1,-46.93429242256052,-76.79304782852142L-26.07460690142251,-42.662804349178565A50,50,0,0,0,-49.98437581378513,-1.2498697957356282Z"
          />
          <path
            className="Gauge-gaugeColor2-3"
            d="M-43.03758404225278,-79.04281346084557A90,90,0,0,1,43.03758404225268,-79.04281346084562L23.9097689123626,-43.912674144914234A50,50,0,0,0,-23.909768912362658,-43.9126741449142Z"
          />
          <path
            className="Gauge-gaugeColor3-4"
            d="M46.934292422560425,-76.79304782852148A90,90,0,0,1,90,-1.0197970015766364e-13L50,-5.6655388976479795e-14A50,50,0,0,0,26.07460690142246,-42.6628043491786Z"
          />
          <circle
            className="needle-center"
            cx={0}
            cy={0}
            r={15}
          />
          <path
            className="needle"
            d="M -9.18485099360515e-16 15 L -70 0 L -9.18485099360515e-16 -15"
          />
        </g>
      </svg>
    </div>
  </div>
</div>
`;
